lib/ledger_sync/adaptors/quickbooks_online/expense/operations/upsert.rb

```diff
@@ -19,51 +19,27 @@ class Contract < LedgerSync::Adaptors::Contract
               end
             end
 
-            private
-
             def build
-              op = if qbo_expense?
-                     Update.new(adaptor: adaptor, resource: resource)
-                   else
-                     Create.new(adaptor: adaptor, resource: resource)
-                   end
-
-              build_account_operation(resource.account)
-              resource.line_items.each do |line_item|
-                build_account_operation(line_item.account)
+              op = if find_result.success?
+                Update.new(adaptor: adaptor, resource: resource)
+              else
+                Create.new(adaptor: adaptor, resource: resource)
               end
-              build_vendor_operation
-              add_root_operation(op)
-            end
-
-            def build_account_operation(account)
-              account_op = Account::Operations::Upsert.new(
-                adaptor: adaptor,
-                resource: account
-              )
+              op.build
 
-              add_before_operation(account_op)
+              op.before_operations.each {|o| add_before_operation(o)}
+              op.after_operations.each {|o| add_after_operation(o)}
+              add_root_operation(op.root_operation)
```

Comment:
The operation class is `Upsert`, and even previously we would just add before/after operations defined in `Upsert` class and then set `root_operation` either `Create`/`Update`

Reply:
I'm not fan of this, but we need to somehow get operations from final operation to current operation..